docs/modules/general/pages/bootstrapping.adoc

```diff
@@ -0,0 +1,567 @@
+* example_base_path - base path of the example Microfrontend (configured via OneCX Core Applications), e.g. `example`.
+--
++
+Prior to routing within Microfrontend's module, the Shell uses shell_url, workspace_name and example_base_path parts of the URL to load the module. Because of this fact, the module's router needs to remove those parts from consideration when matching its routes. Usually, the `path` property of the route is used to control the route activation, but in that case the Microfrontend's module needs a way to only match the relevant part of the URL.
++
+// TODO: Add link to @onecx/angular-webcomponents functions
+// TODO: Add link to startsWith
+// TODO: Add link to initializeRouter
+Using the `startsWith` function from `@onecx/angular-webcomponents`, for https://angular.dev/api/router/UrlMatcher[matcher] property of a route object, results in the router considering only those URL parts relevant to the module. In order for it to work properly, the `initializeRouter` provider has to be added for the module as an app initializer.
++
+During module creation `initializeRouter`:
++
+--
+// TODO: Link to Topic
+* adds Microfrontend information (based on CurrentMfeTopic) to each route.
+* rewrites routes containing `redirectTo` for correct redirection.
+* creates a new route (used when routing away from the module):
+** matched when none of the defined routes were matched.
+** displays nothing (for a period of time when the user routes between Microfrontends).
+--
++
+// TODO: link to statsWith
+The `startsWith` method uses Microfrontend information, saved in the route's data, to remove already used parts from consideration when matching routes within the module.
++
+// TODO: link to lib
+To create your own matchers, please consider using `@onecx/angular-webcomponents` library.
++
+For technologies other than Angular, it is recommended to:
++
+--
+* Use Microfrontend information from CurrentMfeTopic to only use relevant parts of the URL for routing and redirect correctly.
+* Ensure routing away from the module is not causing side-effects.
+--
+
+Configuration::
+All services utilizing HttpClient used within the Microfrontend's module need to know how to make requests to external resources. Depending on the configuration of the Workspace, they need to take that context into consideration when creating a URL for those resources.
++
+A service might want to call `deployment_url/bff/search` by default. With this call being made, MFE App will need to access the BFF. When the Application's path within the Workspace is `mfe/example` the call has to be made to `deployment_url/mfe/example/bff/search`.
++
+// TODO: Link to Topic
+The `apiConfigProvider` presented in the example is utilizing `PortalApiConfiguration` class as `Configuration` for the services. It is listening to the CurrentMfeTopic changes and overwriting the basePath accordingly to the received message and all services use that information to construct a valid URL.
```

Comment:
The `apiConfigProvider` presented in the example is utilizing **the** `PortalApiConfiguration` class as `Configuration` for the services.